src/core/Akka/Actor/FSM.cs

```diff
@@ -151,23 +202,39 @@ public abstract class Reason { }
         /// <summary>
         /// Default <see cref="Reason"/> if calling Stop().
         /// </summary>
-        public class Normal : Reason { }
+        public sealed class Normal : Reason
+        {
+            private Normal() { }
+
+            /// <summary>
+            /// Singleton instance of Normal
+            /// </summary>
+            public static Normal Instance { get; } = new Normal();
+        }
 
         /// <summary>
         /// Reason given when someone as calling <see cref="Stop"/> from outside;
         /// also applies to <see cref="ActorSystem"/> supervision directive.
         /// </summary>
-        public class Shutdown : Reason { }
+        public sealed class Shutdown : Reason
+        {
+            private Shutdown() { }
```

Comment:
This one could break compatibility. Probably I should keep a public constructor with `ObsoletedAttribute`